code/modules/projectiles/guns/energy/laser.dm

```diff
@@ -138,3 +139,95 @@
 
 /obj/item/gun/energy/laser/redtag/hitscan
 	ammo_type = list(/obj/item/ammo_casing/energy/laser/redtag/hitscan)
+
+
+//////////////EXTERNAL LENS////////////////////////////
+
+/obj/item/external_lens
+	name = "external lens"
+	icon_state = "external"
+	desc = "external lens."
+	icon = 'icons/obj/guns/energy.dmi'
+	var/stored_ammo_type = /obj/item/ammo_casing/energy/laser
+	var/overlay = "laser"
+
+/obj/item/external_lens/Initialize()
+	. = ..()
+	add_overlay(overlay)
+
+/obj/item/external_lens/afterattack(atom/movable/AM, mob/user, flag)
+	. = ..()
+	if(istype(AM, /obj/item/gun/energy/laser))
+		var/obj/item/gun/energy/laser/L = AM
+		if(!L.modifystate)
+			L.AddComponent(/datum/component/extralasers, stored_ammo_type, src)
+			forceMove(L)

```

Comment:
Make the lens get deleted when applying it and make a new one from the component when it's removed

Reply:
Don't resolve things that aren't resolved